utils/utils.ts

```diff
@@ -4,30 +4,81 @@
 
 interface IOptions {
   text: string;
-  options: Array<{text:string; correct: boolean}>;
+  options: Array<{ text: string; correct: boolean }>;
 }
 
-export function grade(optionsQuestion: Record<string, boolean>[], answers: IOptions[]) {
-  let notaAcumulada = 0;
-  let allQuestionsRight = true
+export function grade(
+  optionsQuestion: Record<string, boolean>[],
+  answers: IOptions[]
+) {
+  let finalGrade = 0;
+  let allQuestionsRight = true;
 
-  if (optionsQuestion.length !== answers.length) { return "error" }
+  console.log("user", optionsQuestion);
+  console.log("answers", answers);
+
+  if (optionsQuestion.length !== answers.length) {
```

Comment:
maybe this validation can be above before everything else

Reply:
and maybe you can throw an Error, and turn this function into a promise that resolves when you get final result